public/app/plugins/datasource/grafana-azure-monitor-datasource/resourcePicker/resourcePickerData.ts

```diff
@@ -68,14 +68,15 @@ export default class ResourcePickerData extends DataSourceWithBackend<AzureMonit
 
     return resources.map((subscription) => ({
       name: subscription.subscriptionName,
-      id: subscription.subscriptionId,
+      id: `/subscriptions/${subscription.subscriptionId}`,
       typeLabel: 'Subscription',
       type: ResourceRowType.Subscription,
       children: [],
     }));
   }
 
-  async getResourceGroupsBySubscriptionId(subscriptionId: string) {
+  async getResourceGroupsBySubscriptionId(subscriptionURI: string) {
+    const subscriptionId = subscriptionURI.includes('/subscriptions/') ? subscriptionURI.slice(15) : subscriptionURI;
```

Comment:
this is also ambiguous because `subscriptionURI` can be an URI or an ID. Can we make this two functions if needed? (`getResourceGroupsBySubscriptionId` and `getResourceGroupsBySubscriptionURI`) or make it so it's always a URI

Reply:
I am trying to make it so that it's always a URI but users could potentially have broken dashboards where it is not saved as a uri, so this is to try handle that. All new dashboards would have it as a uri. I suppose I could just make the assumption that no one is saving a broken dashboard and expecting it to work and just always assume it's a uri?

Reply:
got it, if that's the case, maybe a comment explaining that should be enough (and a test! so we don't break it unintentionally)